src/audio/codec_adapter/codec_adapter.c

```diff
@@ -236,20 +237,53 @@ int codec_adapter_prepare(struct comp_dev *dev)
 		i++;
 	}
 
-	/* Allocate local buffer */
+	/* allocate buffer for all sinks */
+	if (list_is_empty(&mod->sink_buffer_list)) {
+		for (i = 0; i < mod->num_output_buffers; i++) {
```

Comment:
does this mean it is no longer up to the IPC messages to allocate the buffers?

Reply:
@cujomalainey sorry not following the question. Which buffers are you alluding to? This list of buffers is a replacement for mod->local_buff which assumes there's only one sink for a module. So we expand it to a list to support multiple sinks

Reply:
> 

what does mean "support multiple sinks".?

Reply:
@lenghonglin imagine a demux where you can have a single source split and copied to multiple sinks. Thats is what I mean by "support multiple sinks"

Reply:
got it